CMakeLists.txt

```diff
@@ -7,12 +7,9 @@ set(LOGO [=[
 ]=])
 message(${LOGO})
 
-set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14" CACHE STRING "Minimum OS X deployment version")
-
-project(pennylane_lightning)
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum OS X deployment version")
```

Comment:
Since we are deprecating 10.14 support, we should also make sure to mention this as a breaking change in the changelog.